File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxprofit = 0; 
        int l = 0; 
        int r = 1;
        while (r < prices.size()) {
            if (prices[l] > prices[r]) l = r;
            else maxprofit = max(maxprofit, prices[r] - prices[l]);
            r++;

        }
        return maxprofit;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0
        max1 = 0
        charSet = set()
        for r in range(len(s)):
            while s[r] in charSet: #while it's in charSet, keep removing from left until its gone
                charSet.remove(s[l])
                l+=1              #this keeps the charSet start aligned with the string
            charSet.add(s[r])
            max1 = max(max1, r - l + 1)  #max of curr max and the current substring length (r - l + 1)
        
        return max1
